src/contract_error.ts

```diff
@@ -0,0 +1,77 @@
+import { xdr } from 'stellar-sdk';
+
+export class ContractError extends Error {
+  public type?: string;
```

Comment:
Can this be typed as the enum?

Reply:
We can but because ts enums are numbers, it leads to an extra step for consumers who use the field. Accessing type returns  a number which then needs to be indexed in the enum to retrieve the error. ie `ContractErrorType[ContractError.type]`.  I am not opposed to this but it's worth weighing against not having the type field being `ContractErrorType`